frontend/src/components/movies/MovieDetails.js

```diff
@@ -8,271 +8,222 @@ import { popupNotification } from '../../lib/notification'
 
 import { FaHeart } from 'react-icons/fa'
 
-class MovieDetails extends React.Component {
-  state = {
-    movie: null,
-    formdata: {
-      text: '',
-      rating: ''
-    },
-    liked_movies: '',
-    currentUserId: '',
-    liked_by: '', 
-    heartColor: 'grey',
-    liked: false,
-    commentErrors: {}
-  }
-  
+const MovieDetails = (props) => {
+  const { match } = props
 
-  
-  async componentDidMount() {    
-    // request single movie by id
-    const movieId = this.props.match.params.id
-    // console.log(this.props, movieId)
-    const response = await getSingleMovie(movieId)
-    // console.log(response.data)
-    this.setState({
-      movie: response.data
-    })  
+  const [movie, setMovie] = useState('')
+  const [formData, setFormdata] = useState({ text: '', rating: '' })
+  const [currentUserId, setCurrentUserId] = useState('')
+  const [heartColor, setHeartColor] = useState('grey')
+  const [liked, setLiked] = useState(false)
+
+  useEffect(() => {
+    const getData = async () => {
+      const movieId = match.params.id
+      const response = await getSingleMovie(movieId)
+      const movie = response.data
+      setMovie(movie)
+      // console.log({ movie, response })
 
-    // Get user profile details and update state
-    try {
       const profile = await getUserProfile()
-      // console.log(profile.data.id)
       const userId = profile.data.id
-      // console.log(userId)
-      this.setState({ 
-        currentUserId: userId
-      })
-    } catch (err) {
-      // console.error(err)
-    }
-
-    const likedByArrayIds = this.state.movie.liked_by.map(user => user.id)
-    const isLikedByCurrentUser = likedByArrayIds.includes(this.state.currentUserId)
+      setCurrentUserId(userId)
 
-    const heartColor = isLikedByCurrentUser ? 'crimson' : 'grey'
-
-    this.setState({
-      heartColor,
-      liked: isLikedByCurrentUser
-    })   
-  }
+      const likedByArrayIds = movie.liked_by.map(user => user.id)
+      const isLikedByCurrentUser = likedByArrayIds.includes(currentUserId)
+      setLiked(isLikedByCurrentUser)
+      
+      const heartColor = isLikedByCurrentUser ? 'crimson' : 'grey'
+      setHeartColor(heartColor)
+    }
+    getData()
+  }, [currentUserId, match])
 
-  // function that handles the comments formdata, 
-  handleChange = event => {
-    // console.log(event.target)
-    const formData = {
-      ...this.state.formData,
-      [event.target.name]: event.target.value
-    } 
-    this.setState({ formData })
+  const handleChange = event => {
```

Comment:
`useCallback` ... additionally, you should be more specific with your naming, e.g.: `handleFormDataChange`